master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -836,7 +838,13 @@ private[celeborn] class Master(
     val numReducers = requestSlots.partitionIdList.size()
     val shuffleKey = Utils.makeShuffleKey(requestSlots.applicationId, requestSlots.shuffleId)
 
-    val availableWorkers = workersAvailable(requestSlots.excludedWorkerSet)
+    var availableWorkers = workersAvailable(requestSlots.excludedWorkerSet)
+    if (requestSlots.tagsExpr.nonEmpty) {
+      availableWorkers = tagsManager.getTaggedWorkers(
+        requestSlots.tagsExpr,
```

Comment:
Seems tagsExpr has  a comma-separated list tags

Reply:
@RexXiong Yes, plan for V0 is to take tags input as comma separated string and select workers which have all tags present. (Basically an "AND" op of tags). In later version this can evolve into more complex expression.